SharpPcap/LibPcap/CaptureFileWriterDevice.cs

```diff
@@ -144,31 +144,31 @@ public override void Open(DeviceConfiguration configuration)
         /// </summary>
         /// <param name="p">P.</param>
         /// <param name="h">The height.</param>
-        public void Write(byte[] p, ref PcapHeader h)
+        public void Write(ReadOnlySpan<byte> p, ref PcapHeader h)
         {
             ThrowIfNotOpen("Cannot dump packet, device is not opened");
             if (!DumpOpened)
                 throw new DeviceNotReadyException("Cannot dump packet, dump file is not opened");
 
-            //Marshal packet
-            IntPtr pktPtr;
-            pktPtr = Marshal.AllocHGlobal(p.Length);
-            Marshal.Copy(p, 0, pktPtr, p.Length);
-
             //Marshal header
             IntPtr hdrPtr = h.MarshalToIntPtr();
 
-            LibPcapSafeNativeMethods.pcap_dump(m_pcapDumpHandle, hdrPtr, pktPtr);
-
-            Marshal.FreeHGlobal(pktPtr);
+            unsafe
```

Comment:
does the use of unsafe alter how users can work with the library vs. using the Marshal methods?

Reply:
User wise it is the same, nothing changes.
Logic wise we are temporarily pinning the packet bytes in memory instead of copying them.